src/predict_request_validation_utils.cpp

```diff
@@ -401,7 +402,7 @@ Status RequestValidator<KFSRequest, KFSTensorInputProto, KFSInputTensorIteratorT
         SPDLOG_DEBUG("[servable name: {} version: {}] Invalid batch size - {}", servableName, servableVersion, details);
         return Status(StatusCode::INVALID_BATCH_SIZE, details);
     }
-    if (servableBatchSize.match(rsi.getDim(0))) {
+    if (servableBatchSize.match(rsi.getDim(0)) && rsi.getDim(0) == proto.contents().bytes_contents_size()) {
```

Comment:
This is change has side effects.
With incorrect request request (example shape [5], number of strings: 4, endpoint batch size=4) user can reload model to BS=5. Then proceeding with inference we do not validate the request anymore possibly leading to crash